Classes/common/CDTFetchChanges.h

```diff
@@ -0,0 +1,175 @@
+//
+//  CDTFetchChanges.h
+//  CloudantSync
+//
+//  Created by Michael Rhodes on 31/03/2015.
+//  Copyright (c) 2015 IBM Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+@class CDTDatastore;
+@class CDTDocumentRevision;
+
+/**
+ CDTFetchChanges objects report changes to a database, that is, documents which have been
+ created, updated or deleted. This feature can be used to keep external datastores up to
+ date with changes in a database, for example to keep an external index of documents up
+ to date (we use CDTFetchChanges in the Query implementation, for example).
+
+ CDTFetchChanges provides callbacks for updated and created documents (documentChangedBlock)
+ and deleted documents (documentWithIDWasDeletedBlock). It also provides a completion block
+ for when all changes have been received and processed. The documentChangedBlock is passed the
+ current revision of each document, the documentWithIDWasDeletedBlock is just passed the
+ id of the deleted document.
+
+ In addition, CDTFetchChanges is able to report on changes since a given point: the completion
+ block provides a sequence value which can be passed to future requests to have only changes
+ from that point to be reported. The sequence value should be treated as an opaque value, but
+ can be saved and used across application sessions. This allows for incremental updates in
+ response to changes in a database.
+
+ The CDTFetchChanges object reports all changed documents since the sequence value assigned
+ before -start is called.
+
```

Comment:
Suggested paragraph:

In addition, CDTFetchChanges is able to report on changes since a particular update to the database. Each change made to a CDTDatastore database is assigned a "sequence value". By setting the startSequenceValue, the CDTFetchChanges object will only report all changed documents since that update. This allows you to obtain changes to the database from a more recent update rather than receiving all changes to the database since its creation. Thus, your application can more easily respond to incremental changes in a database.

A sequence value can be obtained from the completion block, fetchRecordChangesCompletionBlock. When the completion block is called, the sequence value associated with the most recent update for the database is passed in via an argument (see below: newSequenceValue). This value can be used for future requests by assigning that value to startSequenceValue before calling -start. Then only changes from that point be reported. 

The sequence value should be treated as an opaque value, but can be saved and used across application sessions.